bridges/zap/zap_encoder.go

```diff
@@ -0,0 +1,211 @@
+package zap
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"go.opentelemetry.io/otel/log"
+	"go.uber.org/zap/zapcore"
+)
+
+// not optimized yet
+// this file implements object and array encoder - similar to memory encoder by zapcore
+var (
+	_ zapcore.ObjectEncoder = (*OtelObjectEncoder)(nil)
+	_ zapcore.ArrayEncoder  = (*sliceArrayEncoder)(nil)
+)
+
+type OtelObjectEncoder struct {
+	// Fields contains the entire encoded log context.
+	Fields []log.KeyValue
+	// cur is a pointer to the namespace we're currently writing to.
+	cur []log.KeyValue
+
+	reflectval log.Value
+	zapcore.Encoder
+}
+
+// NewMapObjectEncoder creates a new map-backed ObjectEncoder.
+func NewOtelObjectEncoder() *OtelObjectEncoder {
+	m := make([]log.KeyValue, 0)
+	return &OtelObjectEncoder{
+		Fields: m,
+		cur:    m,
+	}
+}
+
+// AddArray implements ObjectEncoder.
+func (m *OtelObjectEncoder) AddArray(key string, v zapcore.ArrayMarshaler) error {
+	fmt.Println(v, "inside array")
+	arr := &sliceArrayEncoder{elems: make([]log.Value, 0)}
+	err := v.MarshalLogArray(arr)
+	m.cur = append(m.cur, log.Slice(key, arr.elems...))
```

Comment:
Appending like this is going to require heap allocations to increase the size of the slice. Is this done in the "hot path" of recording logs? If so we need to consider alternate approaches for performance.

Reply:
there is no way to know about array size here - can we agree upon a minimum buffer size?

Reply:
@pellared